components/style/properties.mako.rs

```diff
@@ -5724,6 +5724,97 @@ pub struct PropertyDeclarationBlock {
     pub normal: Arc<Vec<PropertyDeclaration>>,
 }
 
+impl PropertyDeclarationBlock {
+    // https://drafts.csswg.org/cssom/#serialize-a-css-declaration-block
+    pub fn serialize(&self) -> String {
+
+        let mut result_list = String::new();
+        let mut already_serialized = HashSet::new();
+
+        for declaration in self.normal.iter().chain(self.important.iter()) {
+            let property = declaration.name().to_string();
+            if already_serialized.contains(&property) {
+                continue;
+            }
+
+
+            let mut shorthands = Vec::from(declaration.shorthands());
+
+            if shorthands.len() > 0 {
+                shorthands.sort_by(|a,b| a.longhands().len().cmp(&b.longhands().len()));
+
+                // TODO: Help -  I do not want to do iter cloned, is there a way to
+                // somehow collection into a list or Vec of references?
```

Comment:
These are all the errors. The Array remove one makes sense, since you cannot remove from a fixed array: 

```
18:22: 28518:29 error: the trait `core::marker::Sized` is not implemented for the type `[properties::PropertyDeclaration]` [E0277]
/home/divad/Rust/servo/target/debug/build/style-f9fb073aea07cada/out/properties.rs:28518                     .collect::<[PropertyDeclaration]>();
                                                                                                              ^~~~~~~
/home/divad/Rust/servo/components/style/lib.rs:75:5: 75:58 note: in this expansion of include!
/home/divad/Rust/servo/target/debug/build/style-f9fb073aea07cada/out/properties.rs:28518:22: 28518:29 help: run `rustc --explain E0277` to see a detailed explanation
/home/divad/Rust/servo/target/debug/build/style-f9fb073aea07cada/out/properties.rs:28518:22: 28518:29 note: `[properties::PropertyDeclaration]` does not have a constant size known at compile-time
/home/divad/Rust/servo/target/debug/build/style-f9fb073aea07cada/out/properties.rs:28518:22: 28518:29 error: the trait `core::iter::FromIterator<properties::PropertyDeclaration>` is not implemented for the type `[properties::PropertyDeclaration]` [E0277]
/home/divad/Rust/servo/target/debug/build/style-f9fb073aea07cada/out/properties.rs:28518                     .collect::<[PropertyDeclaration]>();
                                                                                                              ^~~~~~~
/home/divad/Rust/servo/components/style/lib.rs:75:5: 75:58 note: in this expansion of include!
/home/divad/Rust/servo/target/debug/build/style-f9fb073aea07cada/out/properties.rs:28518:22: 28518:29 help: run `rustc --explain E0277` to see a detailed explanation
/home/divad/Rust/servo/target/debug/build/style-f9fb073aea07cada/out/properties.rs:28518:22: 28518:29 note: a collection of type `[properties::PropertyDeclaration]` cannot be built from an iterator over elements of type `properties::PropertyDeclaration`
/home/divad/Rust/servo/target/debug/build/style-f9fb073aea07cada/out/properties.rs:28516:21: 28516:34 error: the trait `core::marker::Sized` is not implemented for the type `[properties::PropertyDeclaration]` [E0277]
/home/divad/Rust/servo/target/debug/build/style-f9fb073aea07cada/out/properties.rs:28516                 let mut longhands = self.normal.iter().chain(self.important.iter()).cloned()
                                                                                                             ^~~~~~~~~~~~~
/home/divad/Rust/servo/components/style/lib.rs:75:5: 75:58 note: in this expansion of include!
/home/divad/Rust/servo/target/debug/build/style-f9fb073aea07cada/out/properties.rs:28516:21: 28516:34 help: run `rustc --explain E0277` to see a detailed explanation
/home/divad/Rust/servo/target/debug/build/style-f9fb073aea07cada/out/properties.rs:28516:21: 28516:34 note: `[properties::PropertyDeclaration]` does not have a constant size known at compile-time
/home/divad/Rust/servo/target/debug/build/style-f9fb073aea07cada/out/properties.rs:28516:21: 28516:34 note: all local variables must have a statically known size
/home/divad/Rust/servo/target/debug/build/style-f9fb073aea07cada/out/properties.rs:28578:39: 28578:45 error: no method named `remove` found for type `[properties::PropertyDeclaration]` in the current scope
/home/divad/Rust/servo/target/debug/build/style-f9fb073aea07cada/out/properties.rs:28578                             longhands.remove(index);
                                                         
```